src/odb/include/odb/db.h

```diff
@@ -1886,6 +1886,16 @@ class dbBTerm : public dbObject
 
   uint32_t staVertexId();
   void staSetVertexId(uint32_t id);
+
+  ///
+  /// Set the region where the BTerm is constrained
+  ///
+  void setConstraintRegion(const std::pair<Point, Point>& constraint_region);
+
+  ///
+  /// Get the region where the BTerm is constrained
+  ///
+  std::pair<Point, Point> getConstraintRegion();
```

Comment:
If there is no constraint then you get back (-1,-1),(-1,-1) as a special value which is non-obvious.  It would be cleaner to return a std::optional<std::pair<Point, Point>>